addons/website_sale/controllers/main.py

```diff
@@ -160,13 +160,22 @@ def _get_search_order(self, post):
         # id is added to be sure that order is a unique sort key
         return 'is_published desc,%s , id desc' % post.get('order', 'website_sequence desc')
 
-    def _get_search_domain(self, search, category, attrib_values):
+    def _get_search_domain(self, search, category, attrib_values, search_in_description):
         domain = request.website.sale_product_domain()
         if search:
             for srch in search.split(" "):
                 domain += [
-                    '|', '|', '|', ('name', 'ilike', srch), ('description', 'ilike', srch),
-                    ('description_sale', 'ilike', srch), ('product_variant_ids.default_code', 'ilike', srch)]
+                    '|',
+                    ('name', 'ilike', srch),
+                    ('product_variant_ids.default_code', 'ilike', srch)
+                ]
+                if search_in_description:
+                    domain += expression.OR([
+                        domain,
+                        ['|',
```

Comment:
You combine your domain with another thanks to a OR then you add the result to the original domain...
So from [A] you go to [A, '|', A, B], which is equal to [A]... so this code does in fact nothing (so I suppose that searching in description do not work with this).

There should not be any `domain += ...`, domain must be the result of using an expression.OR/AND stuff. So here you want to say "my new domain, is the old domain OR another domain":
```python
domain = expression.OR([
    domain,
    ['|', (...), (...)]
])
```
and the domain you get from each search term, you want to combine it with the original domain with a "and". So:
```python
domain = expression.AND([
    domain,
    domain2
])
```
In fact, by including all subdomains using `expression`:
```python
domains = [...]
if search:
   for ...:
       subdomains = [
           A,
           B
       ]
       if ...:
           subdomains.append(C)
           subdomains.append(D)

       domains.append(expression.OR(subdomains))
else:
     ...
domain = expression.AND(domains)
```